src/cc65/declare.c

```diff
@@ -517,21 +684,38 @@ static int ParseFieldWidth (Declaration* Decl)
     /* Read the width */
     NextToken ();
     ConstAbsIntExpr (hie1, &Expr);
+    if (!IsClassInt (Decl->Type)) {
+        /* Only integer types may be used for bit-fields */
+        Error ("Bit-field has invalid type '%s'", GetBasicTypeName (Decl->Type));
+        return -1;
+    }
+
+    if (SizeOf (Decl->Type) != SizeOf (type_uint)) {
+        /* Only int sized types may be used for bit-fields for now */
+        Error ("CC65 currently only supports unsigned int bit-fields");
+        return -1;
+    }
+
     if (Expr.IVal < 0) {
         Error ("Negative width in bit-field");
         return -1;
     }
-    if (Expr.IVal > (int) INT_BITS) {
+    /* FIXME: We should compare with the width of the specified type */
+#if 0
+    /* Use is when we really support non-uint16_t bit-fields */
+    if (Expr.IVal > (long)(SizeOf (Decl->Type) * CHAR_BITS)) {
```

Comment:
@acqn As part of the fix for #1095, I have the changes to pass the actual bit-field type around instead of hardcoding uint in various places. That should be ready in a day or two, but let me know if I should split out those changes so you can start working with them.

Reply:
@jmr I am not working on #1047/#1134 as I realized that they are affected by #1095 for their signed part. Would you like to work on them? Anyways please feel free to modify the code here.

Reply:
Yes, I'll work on char/long after signed is working.